onnxscript/converter_test.py

```diff
@@ -580,6 +581,32 @@ def duplicate_output(X):
         outputs = duplicate_output.to_function_proto().output
         self.assertNotEqual(outputs[0], outputs[1])
 
+    def test_bool_attr_promotion(self):
+        @script()
+        def if_then_else(flag: bool, Y, Z):
+            return op.Where(flag, Y, Z)
+
+        @script()
+        def if_then_else_expanded(flag: bool, Y, Z):
+            tmp1 = op.Constant(value_int=flag)
+            tmp2 = op.Cast(tmp1, to=9)
```

Comment:
```suggestion
            tmp2 = op.Cast(tmp1, to=BOOL.dtype)
```

For readability.

We can also implement `__int__` for TensorType I think